compiler/passes/insertLineNumbers.cpp

```diff
@@ -153,15 +152,11 @@ insertLineNumber(CallExpr* call) {
   } else if (fn->hasFlag(FLAG_EXTERN)                           ||
              (fn->hasFlag(FLAG_EXPORT) &&
               !fn->hasFlag(FLAG_INSERT_LINE_FILE_INFO))         ||
-             strcmp(fn->name, "chpl__heapAllocateGlobals") == 0 ||
-             strcmp(fn->name, "chpl__initStringLiterals")  == 0 ||
-             strcmp(fn->name, "chpl__initModuleGuards")    == 0 ||
-             strcmp(fn->name, "chpl_gen_main")             == 0 ||
              ftableMap.count(fn)                                ||
              (mod->modTag == MOD_USER               &&
               !fn->hasFlag(FLAG_COMPILER_GENERATED) &&
               !fn->hasFlag(FLAG_INLINE)) ||
-             (developer && strcmp(fn->name, "halt"))) {
+             (developer && !fn->hasFlag(FLAG_ALWAYS_PROPAGATE_LINE_FILE_INFO))) {
```

Comment:
I think it would be nice if the PR message were clearer that we're adding a flag to allow certain functions to get the same treatment `halt` was already getting.